ckanext/googleanalytics/reports.py

```diff
@@ -3,34 +3,93 @@
 from datetime import datetime, timedelta
 
 
-def google_analytics_dataset_report(last):
+def google_analytics_dataset_report(time):
     '''
     Generates report based on google analytics data. number of views per package
     '''
+    
+    today = datetime.today()
+    if time == 'week':
+        start_date = today - timedelta(days=today.weekday(), weeks=1)
+        end_date = today - timedelta(days=today.weekday() + 1)
+    elif time == 'month':
+        end_date = today.replace(day=1) - timedelta(days=1)
+        start_date = end_date.replace(day=1)
```

Comment:
This should probably have a timedelta of 30 days

Reply:
It returns the previous month. First day of this month - 1 day is the last day of last month. And then start_date is the first day of that month.